x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucket.java

```diff
@@ -216,6 +216,24 @@ private TypeResolution resolveType(BiFunction<Expression, TypeResolutions.ParamO
         return isFoldable(to, sourceText(), FOURTH);
     }
 
+    public static TypeResolution isStringOrDate(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
```

Comment:
Nit, style preference: this could be inlined, potentially by adding an Expression.TypeResolution#or() method, similar to the existing [`and()`](https://github.com/elastic/elasticsearch/blob/a1344106f59a03eb455e73309a537dc2e3133042/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/Expression.java#L53).

But it can also stay as is.

Reply:
I'll keep it here for now, as only AutoBucket uses it. If there is other caller found in the future, factor it into Expression.TypeResolution will be better.